internal/controller/warehouses/git.go

```diff
@@ -169,88 +169,23 @@ func (r *reconciler) selectTagAndCommitID(
 	baseCommit string,
 ) (string, string, error) {
 
-	if sub.CommitSelectionStrategy == kargoapi.CommitSelectionStrategyNewestFromBranch {
-		// In this case, there is nothing to do except return the commit ID at the
-		// head of the branch unless there are includePaths/excludePaths configured to
-		// handle.
-		commit, err := r.getLastCommitIDFn(repo)
+	continueWithoutTag := false
```

Comment:
@maksimstankevic I'm finding the revised logic of this function a bit hard to follow. fwiw, it does look logically correct. At a quick glance, I think just some slight re-ordering of the previous logic might be easier to read. Something along these lines:

```golang
func (r *reconciler) selectTagAndCommitID(
	repo git.Repo,
	sub kargoapi.GitSubscription,
	baseCommit string,
) (string, string, error) {

	var selectedTag, selectedCommit string
	if sub.CommitSelectionStrategy == kargoapi.CommitSelectionStrategyNewestFromBranch {
		// Set selectedCommit
	} else {
		// Get tags

		// Filter tags

		// Bail if no tags

		switch sub.CommitSelectionStrategy {
		case kargoapi.CommitSelectionStrategyLexical:
			// Set selectedTag
		case kargoapi.CommitSelectionStrategyNewestTag:
			// Set selectedTag
		case kargoapi.CommitSelectionStrategySemVer:
			// Set selectedTag
		default:
			// Unknown strategy
		}

		// Bail if no tag selected

		// Checkout the selected tag

		// Set selectedCommit
	}

	// When we get to here, we have selected the correct commit and, if applicable,
	// correct tag, regardless of what the selection strategy was

	// Bail if selectedCommit == baseCommit

	// Getting diffs since baseCommit
	
	// Apply path filers

	return selectedTag, selectedCommit, nil
}
```

Reply:
done